pkg/build/graph/analysis/bc.go

```diff
@@ -98,3 +102,87 @@ func FindCircularBuilds(g osgraph.Graph) []osgraph.Marker {
 
 	return markers
 }
+
+// FindPendingTags inspects all imageStreamTags that serve as outputs to builds.
+//
+// Precedence of failures:
+// 1. A build config points to the non existent tag but no current build exists.
+// 2. A build config points to the non existent tag but the latest build has failed.
+func FindPendingTags(g osgraph.Graph) []osgraph.Marker {
+	markers := []osgraph.Marker{}
+
+ist:
+	for _, uncastIstNode := range g.NodesByKind(imagegraph.ImageStreamTagNodeKind) {
+		istNode := uncastIstNode.(*imagegraph.ImageStreamTagNode)
+		if bcNode, points := buildPointsToTag(g, uncastIstNode); points && !istNode.Found() {
+			buildNode := latestBuild(g, bcNode)
```

Comment:
nit: rename to `latestBuild`?  That helps remember which build we're attaching the marker to.